Common/include/linear_algebra/blas_structure.hpp

```diff
@@ -143,6 +143,345 @@ class CBlasStructure {
     }
   }
 
+  /*!
+   * \brief tred2
+   * Author:
+   *
+   * Original FORTRAN77 version by Smith, Boyle, Dongarra, Garbow, Ikebe,
+   * Klema, Moler.
+   * C++ version by Aashwin Mishra and Jayant Mukhopadhaya.
+   *
+   * Reference:
+   *
+   * Martin, Reinsch, Wilkinson,
+   * TRED2,
+   * Numerische Mathematik,
+   * Volume 11, pages 181-195, 1968.
+   *
+   * James Wilkinson, Christian Reinsch,
+   * Handbook for Automatic Computation,
+   * Volume II, Linear Algebra, Part 2,
+   * Springer, 1971,
+   * ISBN: 0387054146,
+   * LC: QA251.W67.
+   *
+   * Brian Smith, James Boyle, Jack Dongarra, Burton Garbow,
+   * Yasuhiko Ikebe, Virginia Klema, Cleve Moler,
+   * Matrix Eigensystem Routines, EISPACK Guide,
+   * Lecture Notes in Computer Science, Volume 6,
+   * Springer Verlag, 1976,
+   * ISBN13: 978-3540075462,
+   * LC: QA193.M37
+   *
+   * \param[in,out] V: matrix that needs to be decomposed
+   * \param[in,out] d: holds eigenvalues
+   * \param[in,out] e: work vector
+   * \param[in] n: order of matrix V
+   */
+  template<class Mat, class Vec, class W>
+  static void tred2(Mat& V, Vec& d, W& e, int n) {
```

Comment:
@jayantmukho do you know any simple methods that would be stable enough on 3x3 problems to do what tred2 / tql2 do?
That is the only thing standing in the way of vectorizing the UQ stuff (too many if's in these routines).